subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/verification/verifier/DependencyVerificationConfiguration.java

```diff
@@ -233,7 +233,7 @@ public static class TrustedKey extends TrustCoordinates implements Comparable<Tr
             if (keyId.getBytes(StandardCharsets.US_ASCII).length < 40) {
                 throw new InvalidGpgKeyIdsException(Collections.singletonList(keyId));
             } else {
-                this.keyId = keyId;
+                this.keyId = keyId.toUpperCase();
```

Comment:
❌ Case-changing functions that are not intended to be locale-sensitive should pass `Locale.ROOT` explicitly.